### ISSUE TITLE
Support all signature algorithms

### ISSUE THREAD

**Issue by kpp** [est31/rcgen#60]
See https://tools.ietf.org/id/draft-ietf-tls-tls13-21.html#rfc.section.4.2.3 for a full list of signature algos.

So far missing:

- [ ] PKCS_RSA_SHA384
- [ ] PKCS_RSA_SHA512
- [ ] RSA_PSS_SHA256
- [ ] RSA_PSS_SHA384
- [ ] RSA_PSS_SHA512
- [ ] PKCS_ECDSA_P521_SHA512 (depends on https://github.com/briansmith/ring/issues/824)
- [ ] PKCS_ED448

**Comment by est31** [comment]
The non-RSA based algos are blocked by ring support, while RSA based algos haven't been much of a priority for me because ring has no key generation support. I'm open to adding them though.

**Comment by kpp** [comment]
I see. While there is no key generation support, we can at least load them from a file.

**Comment by est31** [comment]
I've tried adding RSA_PSS support in b2733e86011de5ad08f4e54d2231b103f5d56495 but didn't get it to work. So I made support non-pub for now. See also 46bf2eff0b5f8eac1fb8ddf7cc807e168277e3a4 . One could say it's blocked on https://github.com/briansmith/ring/issues/1353 , but I'm not sure.

I might try to build a remote KeyPair that can ingest openssl keys. Based on that I could build support for the remaining signature algorithms.

**Comment by kpp** [comment]
I created a tool in go to generate certs signed with different algos: https://gist.github.com/kpp/c9c84411e17f4b27dddf0d438b289862#file-main-go with hardcoded `var signatureScheme = tls.PSSWithSHA384`. You can use that. I was never able to generate PSS certs with openssl.

**Comment by djc** [comment]
@Alvenix yeah, that probably makes sense. Would be great if you can submit a PR for this!
